images/linux/scripts/installers/java-tools.sh

```diff
@@ -7,34 +7,55 @@ source $HELPER_SCRIPTS/install.sh
 source $HELPER_SCRIPTS/os.sh
 source $HELPER_SCRIPTS/etc-environment.sh
 
-JAVA_VERSIONS_LIST=$(get_toolset_value '.java.versions | .[]')
-DEFAULT_JDK_VERSION=$(get_toolset_value '.java.default')
-
-# Install GPG Key for Adopt Open JDK. See https://adoptopenjdk.net/installation.html
-wget -qO - "https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public" | apt-key add -
-add-apt-repository --yes https://adoptopenjdk.jfrog.io/adoptopenjdk/deb/
-
-if isUbuntu16 || isUbuntu18 ; then
-    # Install GPG Key for Azul Open JDK. See https://www.azul.com/downloads/azure-only/zulu/
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0xB1998361219BD9C9
-    apt-add-repository "deb https://repos.azul.com/azure-only/zulu/apt stable main"
-fi
-apt-get update
-
-for JAVA_VERSION in ${JAVA_VERSIONS_LIST[@]}; do
-    apt-get -y install adoptopenjdk-$JAVA_VERSION-hotspot=\*
-    echo "JAVA_HOME_${JAVA_VERSION}_X64=/usr/lib/jvm/adoptopenjdk-${JAVA_VERSION}-hotspot-amd64" | tee -a /etc/environment
-done
+JAVA_TOOLCACHE_PATH="$AGENT_TOOLSDIRECTORY/Java_Adoptium_jdk"
+
+createEnvironmentVariable() {
+    local JAVA_VERSION=$1
+    local JAVA_PATH=$2
+
+    local JAVA_HOME_PATH=$JAVA_PATH/Contents/Home
+    if [[ $JAVA_VERSION == $JAVA_DEFAULT ]]; then
+        echo "JAVA_HOME=${JAVA_HOME_PATH}" | tee -a /etc/environment
+    fi
+
+    echo "JAVA_HOME_${JAVA_VERSION}_X64=${JAVA_HOME_PATH}" | tee -a /etc/environment
+}
+
+installJavaFromAdoptOpenJDK() {
+    local JAVA_VERSION=$1
+
+    javaRelease=$(curl -s "https://api.adoptopenjdk.net/v3/assets/latest/${JAVA_VERSION}/hotspot" \
+                | jq -r '[.[] | select(.binary.os=="linux" and .binary.image_type=="jdk" and .binary.architecture=="x64")][0]')
+    archivePath=$(echo $javaRelease | jq -r '.binary.package.link')
+    fullVersion=$(echo $javaRelease | jq -r '.version.semver')
+
+    javaToolcacheVersionPath="$JAVA_TOOLCACHE_PATH/$fullVersion"
+    javaToolcacheVersionArchPath="$javaToolcacheVersionPath/x64"
 
-# Set Default Java version.
-if isUbuntu16; then
-    # issue: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=825987
-    # stackoverflow: https://askubuntu.com/questions/1187136/update-java-alternatives-only-java-but-not-javac-is-changed
-    sed -i 's/(hl|jre|jdk|plugin|DUMMY) /(hl|jre|jdk|jdkhl|plugin|DUMMY) /g' /usr/sbin/update-java-alternatives
-fi
-update-java-alternatives -s /usr/lib/jvm/adoptopenjdk-${DEFAULT_JDK_VERSION}-hotspot-amd64
+    echo "Downloading tar archive $archivePath"
+    download_with_retries $archivePath "/tmp" "OpenJDK$JAVA_VERSION.tar.gz"
+    mkdir -p "$javaToolcacheVersionArchPath"
+    tar -xzf "/tmp/OpenJDK${JAVA_VERSION}.tar.gz" -C $javaToolcacheVersionArchPath --strip-components=1
 
-echo "JAVA_HOME=/usr/lib/jvm/adoptopenjdk-${DEFAULT_JDK_VERSION}-hotspot-amd64" | tee -a /etc/environment
+    COMPLETE_FILE_PATH="$javaToolcacheVersionArchPath/x64.complete"
```

Comment:
Looks like it should be: `javaToolcacheVersionPath` as in the https://github.com/actions/virtual-environments/pull/2883/files PR

Reply:
Thank you, fixed